sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -281,10 +266,53 @@ case class GetArrayItem(child: Expression, ordinal: Expression)
 }
 
 /**
- * Common base class for [[GetMapValue]] and [[ElementAt]].
+ * Common trait for [[GetArrayItem]] and [[ElementAt]].
+ */
+trait GetArrayItemUtil {
+
+  /** `Null` is returned for invalid ordinals. */
+  protected def computeNullabilityFromArray(child: Expression, ordinal: Expression): Boolean = {
+    if (ordinal.foldable && !ordinal.nullable) {
+      val intOrdinal = ordinal.eval().asInstanceOf[Number].intValue()
+      child match {
+        case CreateArray(ar) if intOrdinal < ar.length =>
+          ar(intOrdinal).nullable
+        case GetArrayStructFields(CreateArray(elements), field, _, _, _)
+          if intOrdinal < elements.length =>
+          elements(intOrdinal).nullable || field.nullable
+        case _ =>
+          true
+      }
+    } else {
+      true
+    }
+  }
+}
+
+/**
+ * Common trait for [[GetMapValue]] and [[ElementAt]].
  */
+trait GetMapValueUtil extends BinaryExpression with ImplicitCastInputTypes {
+
+  private val child = left
+  private val key = right
+
+  /** `Null` is returned for invalid ordinals. */
+  protected def computeNullabilityFromMap: Boolean = if (key.foldable && !key.nullable) {
+    val keyObj = key.eval()
+    child match {
```

Comment:
Thank you for updating, @maropu . For this one, can we simplify like the following? We can skipping to use the alias variables.
```scala
-  private val child = left
-  private val key = right
-
   /** `Null` is returned for invalid ordinals. */
-  protected def computeNullabilityFromMap: Boolean = if (key.foldable && !key.nullable) {
-    val keyObj = key.eval()
-    child match {
+  protected def computeNullabilityFromMap: Boolean = if (right.foldable && !right.nullable) {
+    val keyObj = right.eval()
+    left match {
```

Reply:
If you want to keep the clear meaning, you may declare them inside of this function. But, since we are not using `parameters`, I think we already have some assumptions that this is inside `BinaryExpression`.
```scala
protected def computeNullabilityFromMap: Boolean = {
    val child = left
    val key = right
    ....
}
```